### PR TITLE
RTL Conversion - NxModal - RSC-1450

### PR DESCRIPTION
**Pull request by avnvbhatta**
https://issues.sonatype.org/browse/RSC-1450

### CODE REVIEW

File: lib/src/components/NxModal/__tests__/NxModal.test.tsx [hunk 1]
```diff
@@ -4,79 +4,64 @@
...
-import { runTimers } from '../../../__testutils__/rtlUtils';
+import { rtlRender, rtlRenderElement, runTimers, userEvent } from '../../../__testutils__/rtlUtils';
+import { render, within } from '@testing-library/react';
 
 describe('NxModal', function() {
-  const dummyCloseHandler = jest.fn();
   const minimalProps: Props = {
     children: 'A message to show in a modal',
-    onClose: dummyCloseHandler
+    onClose: jest.fn()
```
Comment (EmilyBermudez88) [inline lib/src/components/NxModal/__tests__/NxModal.test.tsx:15]:
```suggestion
    onClose: () => {}
```

File: lib/src/components/NxModal/__tests__/NxModal.test.tsx [hunk 2]
```diff
@@ -4,79 +4,64 @@
...
   });
 
-  it('includes any passed in attributes to the nx-modal div', function() {
-    const nxModal = getModal({ id: 'modal-id', lang: 'en_US' });
-
-    expect(nxModal.find('.nx-modal').prop('id')).toEqual('modal-id');
-    expect(nxModal.find('.nx-modal').prop('lang')).toEqual('en_US');
-  });
+  it('includes any passed in attributes to the modal div', function() {
+    const el = renderEl({ id: 'modal-id', lang: 'en_US' })?.firstElementChild;
```
Comment (EmilyBermudez88) [inline lib/src/components/NxModal/__tests__/NxModal.test.tsx:57]:
```suggestion
    const el = renderEl({ id: 'modal-id', lang: 'en_US' })!.firstElementChild;
```

File: lib/src/components/NxModal/__tests__/NxModal.test.tsx [hunk 3]
```diff
@@ -96,93 +81,83 @@ describe('NxModal', function() {
...
+          user = userEvent.setup();
 
+      quickRender({ onCancel: mockCallBack });
       expect(mockCallBack).not.toHaveBeenCalled();
-      component.simulate('keyDown', createEvent());
+
+      await user.keyboard('[Escape]');
+
       expect(mockCallBack).toHaveBeenCalledTimes(1);
       expect(mockCallBack.mock.calls[0][0].type).toBe('cancel');
```
Comment (EmilyBermudez88) [inline lib/src/components/NxModal/__tests__/NxModal.test.tsx:107]: I saw this in Jest's docs at https://jestjs.io/docs/mock-functions, but curious if you know how this is constructed? I think the first [0] refers to the first call made and the second [0] refers to the argument passed, but I'm just a bit unsure about how it comes up first with `mock.calls`- is there a way in JS/React to parse down into events like that to get the type?

File: lib/src/components/NxModal/__tests__/NxModal.test.tsx [hunk 4]
```diff
@@ -96,93 +81,83 @@ describe('NxModal', function() {
...
-      component.simulate('keyDown', createEvent('q'));
-      component.simulate('keyDown', createEvent('Q'));
       expect(mockCallBack).not.toHaveBeenCalled();
     });
 
-    it('calls stopPropagation and stopImmediatePropagation on Escape keydowns', function() {
-      const component = getModal({ onClose: jest.fn() }),
-          escEvent = createEvent(),
-          otherEvent = createEvent('q');
+    it('executes onCancel method ONLY when pressing ESC key', async function () {
```
Comment (EmilyBermudez88) [inline lib/src/components/NxModal/__tests__/NxModal.test.tsx:124]: does onCancel get called when you close the alert by clicking the close button in the modal footer?

File: lib/src/components/NxModal/__tests__/NxModal.test.tsx [hunk 5]
```diff
@@ -4,79 +4,64 @@
...
 
-    expect(nxModal).toMatchSelector('dialog.nx-modal-backdrop');
-    expect(nxModal.children()).toMatchSelector('div.nx-modal');
+    expect(dialog).toBeInTheDocument();
+    expect(dialog).toBe(view.container.firstElementChild);
   });
 
   it('renders children nodes within the modal', function() {
-    const nxModal = getModal({ children: <div className="bar"/> });
+    const view = quickRender({ children: <div className="bar" data-testid="test-div" /> }),
```
Comment (EmilyBermudez88) [inline lib/src/components/NxModal/__tests__/NxModal.test.tsx:31]:
```suggestion
    const view = quickRender({ children: <div data-testid="test-div" /> }),
```
Probably fine without the className included

File: lib/src/components/NxModal/__tests__/NxModal.test.tsx [hunk 6]
```diff
@@ -96,93 +81,83 @@ describe('NxModal', function() {
...
+    });
   });
 
   it('moves focus back to the previously focused element when closed', async function() {
     function Fixture({ modalOpen }: { modalOpen: boolean }) {
       return (
         <>
-          <button id="test-btn">Test</button>
+          <button data-testid="test-btn">Test</button>
           { modalOpen && <NxModal onCancel={jest.fn()}><button id="cancel-btn">Close</button></NxModal> }
```
Comment (entroform) [inline lib/src/components/NxModal/__tests__/NxModal.test.tsx:161]: I wonder if the `id="cancel-btn"` is needed and if not could be removed.

File: lib/src/components/NxModal/__tests__/NxModal.test.tsx (hunk 4 above)
Comment (avnvbhatta) [reply lib/src/components/NxModal/__tests__/NxModal.test.tsx:124]: > does onCancel get called when you close the alert by clicking the close button in the modal footer? And if that is true, should there be a test proving the modal doesn't close when clicking on the dialog element/modal backdrop?

`onCancel` doesn't get called when a button in the modal footer is clicked.

File: lib/src/components/NxModal/__tests__/NxModal.test.tsx (hunk 6 above)
Comment (avnvbhatta) [reply lib/src/components/NxModal/__tests__/NxModal.test.tsx]: > I wonder if the `id="cancel-btn"` is needed and if not could be removed.

Yup, can certainly be removed. df890b63f1dccfd8a2311c44c231a9bdbacd2ba2

File: lib/src/components/NxModal/__tests__/NxModal.test.tsx (hunk 4 above)
Comment (EmilyBermudez88) [reply lib/src/components/NxModal/__tests__/NxModal.test.tsx:124]: > `onCancel` doesn't get called when a button in the modal footer is clicked.

Thanks for clarifying!

File: lib/src/components/NxModal/__tests__/NxModal.test.tsx (hunk 3 above)
Comment (avnvbhatta) [reply lib/src/components/NxModal/__tests__/NxModal.test.tsx:107]: > I saw this in Jest's docs at https://jestjs.io/docs/mock-functions, but curious if you know how this is constructed? I think the first [0] refers to the first call made and the second [0] refers to the argument passed, but I'm just a bit unsure about how it comes up first with `mock.calls`- is there a way in JS/React to parse down into events like that to get the type?

I think you're right. `mock.calls` looks to be an array of the mock calls made. The call itself is another array that stores the arguments passed into it. I think this is just the structure that jest follows. I think we can just use the Event interface to get the type.

**Review by entroform — APPROVED** [review]
This looks good to me,
just need to change the base branch to `RSC-1438_password-manager-nx-modal`
😸

**Pull request merged**